client/my-sites/themes/controller.js

```diff
@@ -49,27 +45,43 @@ function getProps( context ) {
 	};
 
 	return {
-		title,
 		tier,
 		search: context.query.s,
 		trackScrollPage: boundTrackScrollPage,
 		runClientAnalytics
 	};
-}
+};
 
-function makeElement( ThemesComponent, Head, store, props ) {
-	return(
-		<ReduxProvider store={ store }>
-			<Head title={ props.title } tier={ props.tier || 'all' }>
-				<ThemesComponent { ...omit( props, [ 'title', 'runClientAnalytics' ] ) } />
+const LoggedInHead = ( { children, context: { params: { tier, site_id: siteID } } } ) => (
+	<Head
+		title={ buildTitle(
+			i18n.translate( 'Themes', { textOnly: true } ),
+			{ siteID }
+		) }
+		tier={ tier || 'all' }>
+		{ children }
+	</Head>
+);
+
+// This is generic -- nothing themes-specific in here!
+function makeElement( Component, getProps, Head, sideEffects = function() {} ) {
+	// How do we handle the dispatch?
+	return ( context, next ) => {
```

Comment:
Could we cache by `( route, context )` instead?

Aside, I've been pondering merging `Head` into `Main` for a while now, because of issues like this https://github.com/Automattic/wp-calypso/pull/3420#discussion_r53721640 -- lots of overhead needed to set up stuff inside the middleware, which at `Main` component level is already nicely connected to Redux.